aws/resource_aws_dms_endpoint.go

```diff
@@ -306,6 +306,21 @@ func resourceAwsDmsEndpoint() *schema.Resource {
 							Optional: true,
 							Default:  false,
 						},
+						"data_format": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "csv",
+						},
+						"parquet_timestamp_in_millisecond": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+						"parquet_version": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "parquet-1-0",
```

Comment:
Shouldn't it be parquet_1_0 as per the documentation https://docs.aws.amazon.com/cli/latest/reference/dms/create-endpoint.html?

Reply:
This is a very good point. When using this in the `extra_connection_attributes`, then yes, it works like this. (I have not tried the lowercase version though).

However, when used in the `s3_settings`, it takes this format. I would guess that the lowercase-kebab version also works in the extra connection attributes and AWS normalizes it in the background.

Two things:

```
~ $ aws dms describe-endpoints --query "Endpoints[?EndpointIdentifier=='s3-fdy-sbr2'].S3Settings"
[
    {
        "ServiceAccessRoleArn": "arn:aws:iam::SNIP",
        "ExternalTableDefinition": "",
        "CsvRowDelimiter": "\\n",
        "CsvDelimiter": ",",
        "BucketFolder": "fdy/sbr/",
        "BucketName": "SNIP",
        "CompressionType": "GZIP",
        "DataFormat": "parquet",
        "ParquetVersion": "parquet-2-0", <---------------
        "EnableStatistics": true,
        "ParquetTimestampInMillisecond": true,
        "DatePartitionEnabled": false
    }
]
```

Also, please note how this is tested by the acceptance tests and passes.